application-xpoll-ui/src/main/resources/XPoll/XPollSheet.xml

```diff
@@ -278,15 +278,23 @@
           #set ($foundUser = false)
           #foreach ($voteObj in $doc.getObjects('XPoll.XPollVoteClass'))
             #set ($user = $voteObj.user)
-            #set ($isCurrentUser = $user == $xcontext.user)
+            #set ($isCurrentUser = $user == $services.model.serialize($services.model.resolveDocument($xcontext.user), 'local'))
             #set ($votes = $voteObj.getValue('votes'))
             &lt;tr #if($isCurrentUser)class="active"#end&gt;
               #if ($isCurrentUser)
                 #set ($foundUser = true)
               #end
-              #if ($xwiki.exists($user))
+              ## We check if the saved user exists locally in the subwiki or is a global user
+              #set($tmp = $services.model.serialize($services.model.resolveDocument($user)))
+              #set($completeUser = '')
+              #if ($xwiki.exists($tmp))
+                #set ($completeUser = $tmp)
+              #elseif ($xwiki.exists("$xcontext.mainWikiName:$user"))
+                #set ($completeUser = "$xcontext.mainWikiName:$user")
+              #end
+              #if ($completeUser != '')
```

Comment:
This is not good. You need to resolve the user reference relative to the poll reference and that's all. The local user and the global user can exist at the same time and they are **different** users.

Moreover, as a general remark, the user / document reference **syntax** is an internal technical detail that your code should not be aware of. Only the reference serializer and resolver should know the reference syntax. Your code shouldn't perform any string manipulation of the reference. It should just use the resolver or serializer as needed.